vtr_flow/scripts/run_vtr_flow.pl

```diff
@@ -1687,3 +1711,74 @@ sub calculate_relaxed_W {
 
     return $relaxed_W;
 }
+
+sub cmp_full_vs_incr_STA {
+	# loop indices
+	my @indices = (0..6);
+
+	# default output file names
+	my @default_output_names = (
+			$packing_file,
+			$placement_file,
+			$routing_file,
+			$setup_timing_report,
+			$hold_timing_report,
+			$unconstrained_setup_timing_report,
+			$unconstrained_hold_timing_report
+		);
+
+	# The full STA flow should have already been run
+    # directly rename the output files
+    foreach my $index (@indices) {
+    	system_with_timeout(
+            $move_exec, "move.out",
+            $timeout, $temp_dir,
+            $default_output_names[$index], 
+            "full_" . $default_output_names[$index]
+        );
+    }
+
+    my @incremental_vpr_args = @forwarded_vpr_args;
+    push(@incremental_vpr_args, ("--timing_update_type", "incremental"));
+
+    # run incremental STA flow
+    my $fixed_W_log_file = "vpr.out";
+    $q = run_vpr({
+            arch_name => $architecture_file_name,
+            circuit_name => $benchmark_name,
+            circuit_file => $prevpr_output_file_name,
+            sdc_file => $sdc_file_path,
+            pad_file => $pad_file_path,
+            extra_vpr_args => \@incremental_vpr_args,
+            log_file => $fixed_W_log_file,
+        });
+
+    # Rename the incremental STA output files
+    foreach my $index (@indices) {
+    	system_with_timeout(
+            $move_exec, "move.out",
+            $timeout, $temp_dir,
+            $default_output_names[$index], 
+            "incremental_" . $default_output_names[$index]
+        );
+    }
+
+    #Sanity check that full STA and the incremental STA
+    #produce the same *.net, *.place, *.route files
+    #as well as identical timing report files
+    foreach my $index (@indices) {
+    	my $diff_result = &system_with_timeout(
+            $diff_exec, "diff.out",
+            $timeout, $temp_dir,
+            "full_" . $default_output_names[$index],
+            "incremental_" . $default_output_names[$index]
+        );
+        if ($diff_result ne "success") {
+        	print $index;
+        	return "fail";
```

Comment:
I'd expand this error message to show what files failed the diff.

For instance:
```
return "failed: $full_filename $incr_filename differ"
```